src/prival.c

```diff
@@ -122,17 +113,9 @@ stumpless_prival_from_string( const char *string ) {
   len++;
   len = slen - len;
 
-  // Copy the severity substring to the param buffer
-  param = copy_cstring_length( period + 1, len );
-  if( !param ) {
-    return -1;
-  }
-
-  severity = stumpless_get_severity_enum( param );
-
-  free_mem( param );
+  severity = stumpless_get_severity_enum_from_buffer( period + 1, len );
 
-  if( severity < 0 ) {
+  if( severity < 0 && !stumpless_has_error()) {
```

Comment:
rather than an extra part of this conditional, please separate the "has_error" check so that it is in a separate conditional that simply returns (without raising a new error) if an error is present. As is, this branch is skipped in the memory allocation failure case, and the response is only -1 as a side effect of `get_prival` being the result of a bitwise or with -1 rather than being a true error.